// RUN_PIPELINE_TILL: FRONTEND
// ISSUE: KT-59110
// IGNORE_PHASE_VERIFICATION: invalid code inside annotations

<!WRONG_ANNOTATION_TARGET!>@Target(<!ARGUMENT_TYPE_MISMATCH, ARGUMENT_TYPE_MISMATCH!>{}<!>)<!>
interface SomeInterface {
}
